beancount_interpolate/common.py

```diff
@@ -143,6 +148,7 @@ def distribute_over_period(params, default_date, total_value, config):
     amounts = []
     accumulated_remainder = D(str(0))
 
+    # TODO: In the following `begin_date + i * step` can result in an OverflowError
```

Comment:
please elaborate. Do I understand it correctly?
> in case `begin_date + duration` is so close to overflow, that last `begin_date + i * step` which would be beyond it and thus break the while loop raises OverflowError instead.

Reply:
Yes, this would be a corner case. But theoretically one could also specify a step size of `inf` or something like `1 year / 10000 years` which would lead to an OverflowError. The second case is probably not that common so I think the bigger problem will be the more 'hidden' `OverflowError` when 
`begin_date + i * step < end_date` and `begin_date + i * step <= today_date`
but
`begin_date + (i + 1) * step > MAX_DATE`
This would then lead to an `OverflowError` in 
`tmp_date = begin_date + i * step`

This is however not a problem of the new feature as before there was also the possibility of overflowing the `datetime` objects.

Reply:
I will document the TODO in the source code a little more in detail for future reference.